modules/platforms/dotnet/Apache.Ignite.Core.Tests/Cache/Query/CacheDmlQueriesTestSchema.cs

```diff
@@ -0,0 +1,259 @@
+
+            Sql("CREATE INDEX t1_idx_1 ON " + SCHEMA_NAME_1 + '.' + TBL_NAME + "(s1_val)");
+            Sql("CREATE INDEX t1_idx_1 ON " + SCHEMA_NAME_2 + '.' + TBL_NAME + "(s2_val)");
+            Sql("CREATE INDEX t1_idx_1 ON " + Q_SCHEMA_NAME_3 + '.' + TBL_NAME + "(s3_val)");
+            Sql("CREATE INDEX t1_idx_1 ON " + SCHEMA_NAME_4 + '.' + TBL_NAME + "(s4_val)");
+
+            Sql("SELECT * FROM " + SCHEMA_NAME_1 + '.' + TBL_NAME);
+            Sql("SELECT * FROM " + SCHEMA_NAME_2 + '.' + TBL_NAME);
+            Sql("SELECT * FROM " + Q_SCHEMA_NAME_3 + '.' + TBL_NAME);
+            Sql("SELECT * FROM " + SCHEMA_NAME_4 + '.' + TBL_NAME);
+
+            Sql("SELECT * FROM " + SCHEMA_NAME_1 + '.' + TBL_NAME
+                + " JOIN " + SCHEMA_NAME_2 + '.' + TBL_NAME
+                + " JOIN " + Q_SCHEMA_NAME_3 + '.' + TBL_NAME
+                + " JOIN " + SCHEMA_NAME_4 + '.' + TBL_NAME);
+
+            VerifyTables();
+
+            Sql("DROP TABLE " + SCHEMA_NAME_1 + '.' + TBL_NAME);
+            Sql("DROP TABLE " + SCHEMA_NAME_2 + '.' + TBL_NAME);
+            Sql("DROP TABLE " + Q_SCHEMA_NAME_3 + '.' + TBL_NAME);
+            Sql("DROP TABLE " + SCHEMA_NAME_4 + '.' + TBL_NAME);
+        }
+
+        /// <summary>
+        /// Verify tables.
+        /// </summary>
+        private void VerifyTables()
+        {
+            Sql("SELECT SCHEMA_NAME, KEY_ALIAS FROM SYS.TABLES ORDER BY SCHEMA_NAME", res => {
+                res.Equals(new List<List<object>> {
+                    new List<object>{ SCHEMA_NAME_1, "S1_KEY" },
+                    new List<object>{ SCHEMA_NAME_2, "S2_KEY" },
+                    new List<object>{ SCHEMA_NAME_4, "S4_KEY" },
+                    new List<object>{ SCHEMA_NAME_3, "S3_KEY" }
+                });
+            });
+        }
+
+        /// <summary>
+        /// Get test table name.
+        /// </summary>
+        private string GetTableName(bool withSchema) {
+            string prefix = "";
+
+            if (withSchema)
+                prefix += "PUBLIC.";
+
+            return prefix + TBL_NAME;
+        }
+
+        /// <summary>
+        /// Perform SQL query.
+        /// </summary>
+        private void Sql(string qry)
+        {
+            Sql(qry, null);
+        }
+
+        /// <summary>
+        /// Perform SQL query.
+        /// </summary>
+        private void Sql(string qry, Action<IList<IList<object>>> validator)
+        {
+            IList<IList<object>> res = Ignition
+                .GetIgnite()
+                .GetOrCreateCache<int,int>("TestCache")
+                .Query(new SqlFieldsQuery(qry))
+                .GetAll();
+
+            if (validator != null)
+                validator.Invoke(res);
+        }
+ 
+        /// <summary>
+        /// Generate one-row result set.
+        /// </summary>
+        private static List<List<object>> OneRowList(params object[] vals)
+        {
+            return new List<List<object>>{ new List<object>{ vals } };
```

Comment:
Inner list here will contain the entire `vals` as a single element.
I think it should be `return new List<List<object>>{ vals.ToList() };`